software/wmla120_ansible/validate_firewall_setup.yml

```diff
@@ -2,61 +2,94 @@
 #If a firewall is enabled, the following default ports must be granted access on all management hosts for IBM Spectrum Conductor Deep Learning Impact: 9243, 9280, 5000, 5001, 27017, and 6379. If you change these ports after installation, make sure to update firewall rules accordingly.
 #
 
-- name: Setting Variables
-  set_fact:
-    port_9243: '9243/tcp'
-    port_9080: '9080/tcp'
-    port_5000: '5000/tcp'
-    port_5001: '5001/tcp'
-    port_27017: '27017/tcp'
-    port_6379: '6379/tcp'
+- name: Include vars of envs_spectrum_conductor_dli.yml into the 'ports' variable
+  include_vars:
+    file: envs_spectrum_conductor_dli.yml
+    name: ports
 
-- name: List all host firewall data
-  shell: "firewall-cmd --list-all"
-  delegate_to: localhost
+- name: DEBUG - List imported variables
+  debug:
+    msg: "{{ ports }}"
+    
+- name: Check status of firewall
+  command: "firewall-cmd --state"
   register: host_firewall_status
+  ignore_errors: yes
   become: yes
 
-- name: Checking firewall_status stdout
+- name: List all host firewall data
+  command: "firewall-cmd --list-services"
+  when: host_firewall_status.stderr != 'not running'
+  register: host_firewall_data
+  become: yes
+
+- name: DEBUG - Checking firewall_data stdout
   debug:
-    msg: "{{ host_firewall_status.stdout }}"
+    msg: "{{ host_firewall_data.stdout }}"
+  when: host_firewall_status.stderr != 'not running'
 
-- name: Checking firewall_status for http service enabled
+- name: DEBUG - Checking firewall_data for http service enabled
   debug:
     msg: "{{ host_firewall_status.stdout.find('http') }}"
+  when: host_firewall_status.stderr != 'not running'  
 
-- name: Print success status of Host firewall
-  command: echo "Firewall Exists"
+- name: Print success status of host firewall
+  command: echo "Firewall Exists and http enabled"
   when: host_firewall_status.stdout.find('http') != -1
 
-- name: Else print fail status
-  command: echo "Firewall Does Not Exist"
-  when: host_firewall_status.stdout.find('http') == -1
+- name: Print fail status if firewall not enbaled
```

Comment:
Similarly. A debug task is probably what you want here.

Reply:
okay. Have updated with latest commit .